src/common/io.py

```diff
@@ -1798,13 +1803,18 @@ def integration_point(self, solver = None):
         #Sets the parameters, if any
         if solver and self.options["sensitivities"]:
             self.param_sol += [np.array(solver.interpolate_sensitivity(model.time, 0)).flatten()]
-
-    def simulation_end(self):
-        """
-        The finalize method can be used to for instance close the file.
-        ANd this method is called after the simulation has completed.
-        """
-        pass
+        
+        max_size = self.options.get("result_max_size", None)
+        if max_size is not None:
+            current_size = sys.getsizeof(self.time_sol) + sys.getsizeof(self.real_sol) + \
+                            sys.getsizeof(self.int_sol)  + sys.getsizeof(self.bool_sol) + \
+                            sys.getsizeof(self.param_sol)
+            
+            if current_size > max_size:
+                raise ResultSizeError("Maximum size of the result reached (limit: %g GB) at time t=%g. " 
+                                  "To change the maximum allowed result size, please use the option " 
+                                  "'result_max_size' or consider reducing the number of communication "
+                                  "points alternatively the number of variables to store result for."%(max_size/1024**3, self.model.time))
```

Comment:
```suggestion
                raise ResultSizeError("Maximum size of the result reached (limit: %g GB) at time t=%g. " 
                                  "To change the maximum allowed result size, please use the option " 
                                  "'result_max_size', consider reducing the number of communication "
                                  "points or alternatively the number of variables to store result for."%(max_size/1024**3, self.model.time))
```

minor suggestion; Also to be applied for the other occasions.

Reply:
Fixed.